aiida/repository/backend/abstract.py

```diff
@@ -84,15 +84,33 @@ def put_object_from_file(self, filepath: typing.Union[str, pathlib.Path]) -> str
             return self.put_object_from_filelike(handle)
 
     @abc.abstractmethod
+    def has_objects(self, keys: List[str]) -> List[bool]:
+        """Return whether the repository has an object with the given key.
+
+        :param keys:
+            list of fully qualified identifiers for objects within the repository.
+        :return:
+            list of logicals, in the same order as the keys provided, with value True if the respective
+            object exists and False otherwise.
+        """
+
     def has_object(self, key: str) -> bool:
         """Return whether the repository has an object with the given key.
 
         :param key: fully qualified identifier for the object within the repository.
         :return: True if the object exists, False otherwise.
         """
+        return self.has_objects([key])[0]
+
+    @abc.abstractmethod
+    def list_objects(self) -> Iterable[str]:
+        """Return iterable that yeilds all available objects by key.
```

Comment:
```suggestion
        """Return iterable that yields all available objects by key.
```

Reply:
If you use VS Code, I certainly recommend https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker 😉

Reply:
If I'm developing inside a container, will this extension automatically install inside of it when I attach through the remote explorer or will I have to manually re-install it every time I shutdown and re-start the container?

Reply:
I know with the new vs code devcontainers, you can specify which extensions to load, see e.g. https://github.com/chrisjsewell/aiida-install/blob/56fe6d7398594c8813890652af99c521a77f249c/.devcontainer/devcontainer.json#L41

Reply:
When you are using these you need to create the container through the vscode right? I'm not sure this would work for me because I need to initialize mine using docker-compose (I have 3 containers running for this: rabbitmq, postgres, and the work container).

Reply:
I was actually playing around with that a bit but found, at least on my macbook, it was a bit slow a prone to the container dying:

devcontainer.json

```json
// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
// https://github.com/microsoft/vscode-dev-containers/tree/v0.187.0/containers/docker-existing-dockerfile
// https://github.com/microsoft/vscode-dev-containers/tree/main/containers/python-3-postgres
{
    "name": "AiiDA development environment",
    "dockerComposeFile": "docker-compose.yml",
    "service": "main",
    "workspaceFolder": "/workspace",

	// Set *default* container specific settings.json values on container create.
	"settings": {
        "editor.rulers": [100],
        "python.pythonPath": "/usr/local/bin/python",
		"python.defaultInterpreterPath": "/usr/local/bin/python",
        "python.languageServer": "Pylance",
		"python.linting.enabled": true,
		"python.linting.pylintEnabled": true,
        "python.linting.pylintPath": "/usr/local/py-utils/bin/pylint",
        "python.formatting.provider": "yapf",
        "python.formatting.yapfPath": "/usr/local/py-utils/bin/yapf",
    },
	
	// Add the IDs of extensions you want installed when the container is created.
	"extensions": [
		"ms-python.python",
		"ms-python.vscode-pylance",
		"ms-azuretools.vscode-docker",
		"streetsidesoftware.code-spell-checker",
		"chrisjsewell.aiida-explore-vscode"
	],
}
```

```yaml
version: '3.4'

services:

  main:
    image: 'mcr.microsoft.com/vscode/devcontainers/python:3.8'
    volumes:
      - ..:/workspace:cached
    # Overrides default command so things don't shut down after the process ends.
    command: sleep infinity

  postgres:
    image: postgres:12.3
    environment:
      POSTGRES_USER: pguser
      POSTGRES_PASSWORD: password
    restart: unless-stopped

  rmq:
    image: rabbitmq:3.8.3-management
    environment:
      RABBITMQ_DEFAULT_USER: guest
      RABBITMQ_DEFAULT_PASS: guest
    restart: unless-stopped
```

Reply:
Uhm, very interesting! Thanks, I will give this a try =D